store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -2495,7 +2338,7 @@ public void deleteMailbox(DeleteBlobs deleteBlobs) throws ServiceException {
             boolean needRedo = needRedo(null, redoRecorder);
             boolean success = false;
             try {
-                beginTransaction("deleteMailbox", null, redoRecorder);
+                final MailboxTransaction t = new MailboxTransaction("deleteMailbox", null, l, redoRecorder);
```

Comment:
I did not use `try-with-resources` here because `beginTransaction` occurs, a few other operations occur, and then a try block opens up where the transaction is marked for commit and closed/ended in finally. Beginning the transaction in the try block where it is committed/closed would have changed the logical order of operations for this block of code. There is probably a bug here... but I'm not sure what it is and I'm trying to avoid any change of behavior/logic.

Reply:
I did not use `try-with-resources` here because `beginTransaction` occurs, a few other operations occur, and then a try block opens up where the transaction is marked for commit and closed/ended in finally. Beginning the transaction in the try block where it is committed/closed would have changed the logical order of operations for this block of code. There is probably a bug here... but I'm not sure what it is and I'm trying to avoid any change of behavior/logic.